src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,50 +1,163 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> node = new Node<>(last, value, null);
+        if (size == 0) {
+            first = node;
+        } else {
+            last.next = node;
+        }
+        last = node;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+
+        checkIndex(index);
+        Node<T> oldNode = getNodeByIndex(index);
 
+        Node<T> newNode = new Node<T>(oldNode.prev, value, oldNode);
+        if (index > 0) {
+            newNode.prev.next = newNode;
+        } else {
+            first = newNode;
+        }
+        oldNode.prev = newNode;
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T item: list) {
+            add(item);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+
+        Node<T> node = getNodeByIndex(index);
+        return node.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+
+        Node<T> node = getNodeByIndex(index);;
+        T previosItem = node.item;
+        node.item = value;
+        return previosItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+
+        Node<T> node = getNodeByIndex(index);
+        if (index > 0) {
+            node.prev.next = node.next;
+        } else {
+            first = node.next;
+        }
+        if (node.next != null) {
+            node.next.prev = node.prev;
+        } else {
+            last = node.prev;
+        }
+
+        T removedItem = node.item;
```

Comment:
move the whole logic of changing links to `unlink()` method

Reply:
done